### PR TITLE
Change ApplicationSubscribers nacks to acks

### PR DESCRIPTION
**Pull request by bill-transue**
This will change nacks to acks in all cases.

### CODE REVIEW

**Review by saikumar9 — DISMISSED** [review]
Changed nacks to acks for EventSource Subscribers.

**Review by markgoho — CHANGES_REQUESTED** [review]
please provide a ticket to track this work, tests to cover the change, and a plan for QA

**Comment by mdkaraman** [comment]
@markgoho asked me to take a look at this PR.  In addition to his comment above, which might answer some of the following, I have a couple questions:
- Why are ALL nacks being changed to acks?
- If this is definitely the desired behavior, shouldn't we also be changing this error message to not say "nacked"?
```
logger.debug "application_submitted_subscriber_error: baacktrace: #{e.backtrace}; nacked"
```

**Comment by bill-transue** [comment]
> please provide a ticket to track this work, tests to cover the change, and a plan for QA

I don't have a ticket for this work. This was requested by @ipublic directly, I had originally asked for tickets to be made but he asked for this change direct. QA plan is being handled by Sarah who has a specific list of people who can't make it through this.

**Comment by bill-transue** [comment]
I don't know why we would change the nacks to acks. This change was specifically requested by Dan. I know that @matt--williams is looking for the nack's in logs so I didn't change those in the debug output. QA plan is to submit apps in the spreadsheet listing failures. No ticket was made, this change was requested by Dan.

**Comment by saikumar9** [comment]
@mdkaraman @bill-transue We are changing all **nacks** to **acks** because this is not the right use case to have nacks in the first place. We usually use nacks to put the messages back into the queue to request the external system again but in our case, we are not going to get a different result to put the message back in the queue. "It is also by convention that we do not nack/reject back to EventSource at this stage" per Dan.

**Comment by mdkaraman** [comment]
I see.  Thanks for clarifying, @saikumar9.  For what it's worth, it still seems like we should update the debug output so it better reflects the actual code.  Could cause confusion down the line.  Maybe there is a different word @matt--williams could target?

File: app/event_source/subscribers/application_subscriber.rb [hunk 1]
```diff
@@ -23,11 +23,11 @@ class ApplicationSubscriber
         logger.debug "application_submitted_subscriber_message; acked"
       else
         errors = result.failure.errors.to_h
-        nack(delivery_info.delivery_tag)
+        ack(delivery_info.delivery_tag)
         logger.debug "application_submitted_subscriber_message; nacked due to:#{errors}"
```
Comment (markgoho) [inline app/event_source/subscribers/application_subscriber.rb]: please update logging message to reflect change in acknowledgement type

File: app/event_source/subscribers/application_subscriber.rb [hunk 2]
```diff
@@ -23,11 +23,11 @@ class ApplicationSubscriber
...
       end
     rescue StandardError => e
-      nack(delivery_info.delivery_tag)
+      ack(delivery_info.delivery_tag)
       logger.debug "application_submitted_subscriber_error: baacktrace: #{e.backtrace}; nacked"
```
Comment (markgoho) [inline app/event_source/subscribers/application_subscriber.rb]: please update logging message to reflect change in acknowledgement type

File: app/event_source/subscribers/mec_check_subscriber.rb [hunk 3]
```diff
@@ -22,7 +22,7 @@ class MecCheckSubscriber
           mec_check = Aces::MecCheck.find(mc_id)
           mec_check.update!(failure: error)
         end
-        nack(delivery_info.delivery_tag)
+        ack(delivery_info.delivery_tag)
         logger.debug "application_submitted_subscriber_message; nacked due to: #{error}"
```
Comment (markgoho) [inline app/event_source/subscribers/mec_check_subscriber.rb]: please update logging message to reflect change in acknowledgement type

File: app/event_source/subscribers/mec_check_subscriber.rb [hunk 4]
```diff
@@ -35,7 +35,7 @@ class MecCheckSubscriber
           failure: e
         }
       )
-      nack(delivery_info.delivery_tag)
+      ack(delivery_info.delivery_tag)
       logger.debug "application_submitted_subscriber_error: backtrace: #{e.backtrace}; nacked"
```
Comment (markgoho) [inline app/event_source/subscribers/mec_check_subscriber.rb]: please update logging message to reflect change in acknowledgement type

File: app/event_source/subscribers/transfer_subscriber.rb [hunk 5]
```diff
@@ -34,7 +34,7 @@ class TransferSubscriber
           outbound_transfer = Aces::Transfer.find(transfer_id)
           outbound_transfer.update!(failure: error)
         end
-        nack(delivery_info.delivery_tag)
+        ack(delivery_info.delivery_tag)
         logger.debug "application_submitted_subscriber_message; nacked due to:#{errors}"
```
Comment (markgoho) [inline app/event_source/subscribers/transfer_subscriber.rb]: please update logging message to reflect change in acknowledgement type

File: app/event_source/subscribers/transfer_subscriber.rb [hunk 6]
```diff
@@ -53,7 +53,7 @@ class TransferSubscriber
         inbound_transfer = Aces::InboundTransfer.where(external_id: ib_transfer_id).last
         inbound_transfer&.update!(failure: true)
       end
-      nack(delivery_info.delivery_tag)
+      ack(delivery_info.delivery_tag)
       logger.debug "application_submitted_subscriber_error: baacktrace: #{e.backtrace}; nacked"
```
Comment (markgoho) [inline app/event_source/subscribers/transfer_subscriber.rb]: please update logging message to reflect change in acknowledgement type

**Review by markgoho — CHANGES_REQUESTED** [review]

**Review by markgoho — APPROVED** [review]

**Pull request merged**